src/FileMigrationHelper.php

```diff
@@ -78,6 +78,13 @@ protected function migrateFile($base, File $file, $legacyFilename)
             return false;
         }
 
+        // Remove this file if it violates allowed_extensions
+        $allowed = array_filter(File::config()->allowed_extensions);
+        $extension = strtolower($file->getExtension());
+        if (!in_array($extension, $allowed)) {
+            $file->delete();
+            return false;
+        }
```

Comment:
This seems quite aggressive, given that files could've been added but disallowed some time after.

although I guess editing the file would always have a failed validation, but you wouldn't have lost the file.

Maybe @chillu could provide some insight for the direction of this.

Reply:
Agreed, this does seem aggressive - deleting files shouldn't be something we are doing. If they got there they should stay. Perhaps some fields have different allowed_types per instance that mean files that are ordinarily blocked are allowed under some points of entry...

Reply:
I think that's too agressive as well, thanks for flagging that Chris. There might be viable use cases where files were added via other means to the CMS, but aren't user uploadable (not in allowed_extensions). Also, will this cause error pages (`assets/error-500.html`) to be served dynamically until they're republished manually by CMS authors?

Reply:
This deletes ONLY the record. static files stay in place.

What should happen with the records for these types if they can't be imported, then?

Reply:
@tractorcow perhaps we could have an opt-out for this disallowed file types feature?

The case @dhensby raised is quite valid, although I wonder how common that would be?

Reply:
I could have a config flag to disable deletion; Instead it'd just not migrate those files.

What should the default for this be?

Reply:
we can have it not migrate disallowed_files and there's the ability opt-out of that.